### PR TITLE
feat: addResourceSuppressionsByPath throws error when resource is not found.

### PR DESCRIPTION
**Pull request by maclema**
When calling addResourceSuppressionsByPath, if no resource is found an error is thrown. Calling addResourceSuppressionsByPath and not finding a resource could indicate a problem in the code, and could lead to future unintended suppressions.

### CODE REVIEW

**Review by dontirun — CHANGES_REQUESTED** [review]

File: test/Engine.test.ts [hunk 1]
```diff
@@ -736,6 +736,20 @@ describe('Rule suppression system', () => {
...
+  test('Test path miss', () => {
+    const stack = new Stack();
+    try {
+      NagSuppressions.addResourceSuppressionsByPath(stack, '/No/Such/Path', [
+        { id: 'NA', reason: '............' },
+      ]);
+      throw new Error('Did not fail');
+    } catch (err) {
+      expect(err + '').toBe(
+        'Error: Suppression path did not match any resource'
```
Comment (dontirun) [inline test/Engine.test.ts:749]: Needs to be updated with new Error message

**Review by dontirun — CHANGES_REQUESTED** [review]

**Comment by maclema** [comment]
Confirmed unit tests passing with updated error message.

**Review by dontirun — APPROVED** [review]

**Comment by dontirun** [comment]
Related to #800

**Comment by dontirun** [comment]
@maclema Thanks for the contribution!

**Pull request merged**